From patchwork Fri Apr  2 00:43:27 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Krish Sadhukhan <krish.sadhukhan@oracle.com>
X-Patchwork-Id: 12180181
Return-Path: <kvm-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-18.8 required=3.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,HEADER_FROM_DIFFERENT_DOMAINS,INCLUDES_CR_TRAILER,
	INCLUDES_PATCH,MAILING_LIST_MULTI,MSGID_FROM_MTA_HEADER,SPF_HELO_NONE,
	SPF_PASS,USER_AGENT_GIT autolearn=ham autolearn_force=no version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id 29A6FC43462
	for <kvm@archiver.kernel.org>; Fri,  2 Apr 2021 01:33:06 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id EF2A461001
	for <kvm@archiver.kernel.org>; Fri,  2 Apr 2021 01:33:05 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S234153AbhDBBdD (ORCPT <rfc822;kvm@archiver.kernel.org>);
        Thu, 1 Apr 2021 21:33:03 -0400
Received: from aserp2130.oracle.com ([141.146.126.79]:49780 "EHLO
        aserp2130.oracle.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S233677AbhDBBdA (ORCPT <rfc822;kvm@vger.kernel.org>);
        Thu, 1 Apr 2021 21:33:00 -0400
Received: from pps.filterd (aserp2130.oracle.com [127.0.0.1])
        by aserp2130.oracle.com (8.16.0.42/8.16.0.42) with SMTP id
 1321Wr5m063887;
        Fri, 2 Apr 2021 01:32:53 GMT
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=oracle.com;
 h=from : to : cc :
 subject : date : message-id : in-reply-to : references :
 content-transfer-encoding : content-type : mime-version;
 s=corp-2020-01-29; bh=OyfSkumGwoB6TlE1OOlaXy++iJqR4mB9HcHh/k8EHrE=;
 b=jNo5RH2ixc/1r/s8FKOu+QXCV96GW54FoZtBvowSP3s04vlu7idcXbuuHabJ9dt8+sCI
 7X9GAusT7Ees2hUWk0HtIduMoNP84ujFTwxJlInYcp6eQB6IetR5ZbS7iNYT8gwc7Kbm
 n80Y1zUNmeMRf0brFqwIgk3S6ArvIsajUJk6qQfyFbktiYVjJAu95H/m+JQxYZfOpWGJ
 0lPrC3WT6TSEjjfoVhPx/1SUYUSmq+xQN3+uuHU1Sc0juJmUFsy3ttP70ZCQL2WkoUwz
 vvZsF7J8DlqLr+/ykzo8yIJB8PXBFWbLc7bbqDiRoyUv/pZun23IHH82bFGI3aC0mYmp Ew==
Received: from aserp3020.oracle.com (aserp3020.oracle.com [141.146.126.70])
        by aserp2130.oracle.com with ESMTP id 37n33dugm2-1
        (version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256
 verify=OK);
        Fri, 02 Apr 2021 01:32:53 +0000
Received: from pps.filterd (aserp3020.oracle.com [127.0.0.1])
        by aserp3020.oracle.com (8.16.0.42/8.16.0.42) with SMTP id
 1321Uw7M018855;
        Fri, 2 Apr 2021 01:32:53 GMT
Received: from nam10-mw2-obe.outbound.protection.outlook.com
 (mail-mw2nam10lp2101.outbound.protection.outlook.com [104.47.55.101])
        by aserp3020.oracle.com with ESMTP id 37n2abyytf-2
        (version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256
 verify=OK);
        Fri, 02 Apr 2021 01:32:53 +0000
ARC-Seal: i=1; a=rsa-sha256; s=arcselector9901; d=microsoft.com; cv=none;
 b=PtVB0ILd9sc70XNDsTSuB5AWi30SB2QHc9JCI0is0v9/ArZtLO5v6dyGJbJ19VWnUZH86Vg81U3rW0brur976KdD4naHWFb3Puuf7VKujfgyTyzlUFWKsq8uk7hWRYY8NkawUWsmkgDXKfkHNkgjDBqZpViBoI/YTtKajQ41WG40VNkvO6a45oOf6BEdBjyzrza9jlRcXtNFneExyxqjiEAQKLXplvVWopkZvyaRPaBTfo1OfFDrQuHwTywhBrkbMxgWQMGoXEGSlaQlVk5SJ9qdzKf2myUmNZbpLFDPHfTPQKdTN+sUjF3xuTI0VusgJ5hDp5jfnH3v8VRhgwIudQ==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=microsoft.com;
 s=arcselector9901;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=OyfSkumGwoB6TlE1OOlaXy++iJqR4mB9HcHh/k8EHrE=;
 b=TAII1ryUxAd7qHOXiaV9mai9mVvBJncSiDWE1WgGBrpuZaSZ+qqFLeqOkb4cuAamZphG6o/3G1uBUstwWqjrHziXvNaAoZftSZ+uR46G1ASAabreswkLPGRcdDrGZS544iZFGg2vBKUlD6iY57zMAmc/oiCzYmNx0R7YZ0jFNxcwWX9zxnmAGLAfVXn7G+ngkkQKsnxc3pWbVorhG0D9wedMVodCYlwAjh1JLA/F7MyUab95olncWpRjgndB9TdsrRhroNx0RBbRm3myF4dr26eCrkjkCs3ut09FyYi0irViOty6k9mU+5zC674Hko4TkI9pOdcCmtqT3HCc33XREg==
ARC-Authentication-Results: i=1; mx.microsoft.com 1; spf=pass
 smtp.mailfrom=oracle.com; dmarc=pass action=none header.from=oracle.com;
 dkim=pass header.d=oracle.com; arc=none
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=oracle.onmicrosoft.com; s=selector2-oracle-onmicrosoft-com;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=OyfSkumGwoB6TlE1OOlaXy++iJqR4mB9HcHh/k8EHrE=;
 b=vftsXn7U226XuwWUhlt4tcBQajz+UPm2+IgkEuJqzCjKXp/jTINCzslSwQhmOQ2Ns/PiWdz73fSTRJVdmTLA16UqsOwkrw+YhybLEyj39sTRED5KoLv6ykeJcjDCi8y68kwe1LQPbcZnE6Lo/U+tMk0XKEAkTCXxnPmicnYoisY=
Authentication-Results: vger.kernel.org; dkim=none (message not signed)
 header.d=none;vger.kernel.org; dmarc=none action=none header.from=oracle.com;
Received: from SN6PR10MB3021.namprd10.prod.outlook.com (2603:10b6:805:cc::19)
 by SA2PR10MB4795.namprd10.prod.outlook.com (2603:10b6:806:11d::14) with
 Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.3999.28; Fri, 2 Apr
 2021 01:32:52 +0000
Received: from SN6PR10MB3021.namprd10.prod.outlook.com
 ([fe80::5911:9489:e05c:2d44]) by SN6PR10MB3021.namprd10.prod.outlook.com
 ([fe80::5911:9489:e05c:2d44%5]) with mapi id 15.20.3999.029; Fri, 2 Apr 2021
 01:32:52 +0000
From: Krish Sadhukhan <krish.sadhukhan@oracle.com>
To: kvm@vger.kernel.org
Cc: pbonzini@redhat.com, jmattson@google.com, seanjc@google.com
Subject: [PATCH 1/5 v6] KVM: SVM: Define actual size of IOPM and MSRPM tables
Date: Thu,  1 Apr 2021 20:43:27 -0400
Message-Id: <20210402004331.91658-2-krish.sadhukhan@oracle.com>
X-Mailer: git-send-email 2.25.4
In-Reply-To: <20210402004331.91658-1-krish.sadhukhan@oracle.com>
References: <20210402004331.91658-1-krish.sadhukhan@oracle.com>
X-Originating-IP: [138.3.201.29]
X-ClientProxiedBy: SJ0PR13CA0024.namprd13.prod.outlook.com
 (2603:10b6:a03:2c0::29) To SN6PR10MB3021.namprd10.prod.outlook.com
 (2603:10b6:805:cc::19)
MIME-Version: 1.0
X-MS-Exchange-MessageSentRepresentingType: 1
Received: from ban25x6uut29.us.oracle.com (138.3.201.29) by
 SJ0PR13CA0024.namprd13.prod.outlook.com (2603:10b6:a03:2c0::29) with
 Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.4020.9 via Frontend
 Transport; Fri, 2 Apr 2021 01:32:50 +0000
X-MS-PublicTrafficType: Email
X-MS-Office365-Filtering-Correlation-Id: 36ed3ced-e2d7-41cf-4024-08d8f5773b3c
X-MS-TrafficTypeDiagnostic: SA2PR10MB4795:
X-Microsoft-Antispam-PRVS: 
 <SA2PR10MB4795EBA4073203C944324236817A9@SA2PR10MB4795.namprd10.prod.outlook.com>
X-MS-Oob-TLC-OOBClassifiers: OLM:6790;
X-MS-Exchange-SenderADCheck: 1
X-Microsoft-Antispam: BCL:0;
X-Microsoft-Antispam-Message-Info: 
 JuZLTw5P4O5/6qp/DVwwx664vdRnsZbkL/L7m8PSpAzL5zRd9DGvBALL7Rxl3S275xDvfqgsIB4bYJIxEzMU1qJvaf/qrTnmbMnJCzmZApopeXga6FEwSVfHe3gVuJypcKsmXDUYBXb2m4dE/+cofCC++GwEoyg9S8TBb3M1gvNwo2t+DrlRv0pPOitsmnSm7uX7yYoK6fH3+HqYwgZYbd6WjvR1d9LJv1AoeIhNw9UKsaaRXg6mLcBGqZ3vZHDfw56EtX39WopKf02iAKGrepf+9JZYBCEXJ/VwsF18lIGRhXJ3d5+Vyv3WY4Ttxa1D2VcH74a+NEvcKa53nCZ16CIuIR3oNN2UHyCxgVRpKgW8LcHedI85GpqiPB8owL/9Gu2mE1AweLoRXfF4zqX6MVPdA1oV5WgkirJe573TaVhVuuf8/uPoKpx1apz8ZFdpMu1CNzSDNXpJnVkWGqnzUx1ZeTpGnq3DQQEnDm4TYXlJkxrogvonVcqQqfDrtGlKHFbCabMlD77NE4XBXixvAnq5OGS7Qgz/CJoTeEgbiVDtqNqNz4qPuH3nlU4R/xGZlmwSFXBojw8nb9ibrgw70syJMW8BpUYMX9zyfPbDlvQZ3zNYvydJVeK6bfhoQpDqJYm01uFwmp4a9nQWfADN0L/CIg/X8d2Eg+SWoVeay54=
X-Forefront-Antispam-Report: 
 CIP:255.255.255.255;CTRY:;LANG:en;SCL:1;SRV:;IPV:NLI;SFV:NSPM;H:SN6PR10MB3021.namprd10.prod.outlook.com;PTR:;CAT:NONE;SFS:(376002)(366004)(346002)(396003)(39860400002)(136003)(6666004)(26005)(36756003)(316002)(83380400001)(1076003)(66556008)(5660300002)(16526019)(66476007)(186003)(7696005)(6486002)(2616005)(44832011)(52116002)(956004)(8936002)(38100700001)(8676002)(6916009)(2906002)(66946007)(86362001)(4326008)(478600001);DIR:OUT;SFP:1101;
X-MS-Exchange-AntiSpam-MessageData: 
 EVoIeqw780YxPZHTHElBFwW3Ppf4doLoFXQPuQ9Pd/N9Nsjtoq34t9asadjkK7F4gY2c5R1STvd+lhYVMCVz4CP3p7pAIped+PXFrr79bDJls1D1jN/Tg/4nETPyG5W+wRoOuP0Fgfej2ZUijpUQ3yls6BEbEZx8ySKb2qXSOUCrlcgpWSR3QPEwnA5paFkq69iibo12/lqIh5XvyNf6NYv4BDbpcb/uitvdRsjhwkENco+vN3ujxf2AcUvrQkxz6sDfOc53BmUe+fNqH11G9DqU6ff+WbI08Isawjcnu78o66CTCwAHpxAJf+MdlPbWW7Hi++WZOLucUxojnx50QCMyOXf2u92Ynaf7V9/F5o6ynb6/huawiL4sWg7e4iaAdw2wBvCiftySvrhmsQxZGecCIhCb0PpZMfrrn6hGJejFrfn34dnwdbK/smFNlGspSww7inh5VZ9TYNO8FAfdSiRlJDdMVWcpsTTlP9EHpLVkIopXnVDHmpnzoIoRSc9Rq6BKFVM6pgNB8mwpFnQxtLJmxJyHWVd0Vg7J0Y6NH8gP7DhcMMTpdYu80pNt/yeNYA/ZynKpfm1Qlb4sPCv6F1tRWattUukUbc2x0TczO7AaQT9sE5+kNrLwxLWZaIbSCbK/9l6AnHXg9dJmgHnpOUz3XNE2V3S1Zg9q6sU19zLTUIYpkMGGTNFCiRSPFuyr2sCNMUyse3NRPOaoqTVhcJX8oPysStz90TWesfUt0hR82He1C8JPLSmPYAMr+mmcOJMmALMpPyziHRucBOFznc+HKJJhNXAGB7bpOqUp5S/hKizvCTHNBqhprD5IaT/QO8+DMsn+phNJuKQPYJ/oPZG8gJZnUL5D7nEqoR2B71736dNURzhKAsZxG/1GSBGHnDixb06+8HCVmKWiOK4C1yEYngqh8YU8CtPPCPyyq6a5y+5palhn0H1LRs6PdCy8eGXlWJLWXhASzWNgbApW38z+KEKnlyQPnEk11onYKE4b1bYhuHBZOK96Gno62/AmzSJVvGb4Dl6py8FFyuU0W29h2Bo9D8HRe9td223Fo1heGPFIlIMugeURoLO5MqCC8hdKJXxherPXgA/EXs/a9Dw819ZblIfdipRuZ3217VRnWMQfgzepOSP1/2OjSdsmhkDkVfymewzslHgiD/1LmYMDSpc/CKkxWWv1Xz7smEn4fveTU/zA1KGvso6FGivbXadXgqi21X6w48dMPbWUUvEKNB+HxXfB9LWD2UtU2vMIxw9X0AMmX0QKem6kTy9cgWsT2eMyXXtqpGL+JJUQ8bjgBRJ4VMJotvPSoyAbEF/OOZ3CeLTiziKruTlm3c1V
X-OriginatorOrg: oracle.com
X-MS-Exchange-CrossTenant-Network-Message-Id: 
 36ed3ced-e2d7-41cf-4024-08d8f5773b3c
X-MS-Exchange-CrossTenant-AuthSource: SN6PR10MB3021.namprd10.prod.outlook.com
X-MS-Exchange-CrossTenant-AuthAs: Internal
X-MS-Exchange-CrossTenant-OriginalArrivalTime: 02 Apr 2021 01:32:51.9021
 (UTC)
X-MS-Exchange-CrossTenant-FromEntityHeader: Hosted
X-MS-Exchange-CrossTenant-Id: 4e2c6054-71cb-48f1-bd6c-3a9705aca71b
X-MS-Exchange-CrossTenant-MailboxType: HOSTED
X-MS-Exchange-CrossTenant-UserPrincipalName: 
 dZctJulZQT23TyRuYG3/xOxYJRf7aLKy7tlle0n+5pRzK/PAbszxOGDwPhTg2Nyl98IGATWKjnOxPiGNYpbzbI6mvjWpPzTxkz0WRRHH0Sg=
X-MS-Exchange-Transport-CrossTenantHeadersStamped: SA2PR10MB4795
X-Proofpoint-Virus-Version: vendor=nai engine=6200 definitions=9941
 signatures=668683
X-Proofpoint-Spam-Details: rule=notspam policy=default score=0 adultscore=0
 mlxscore=0 bulkscore=0
 suspectscore=0 phishscore=0 malwarescore=0 mlxlogscore=999 spamscore=0
 classifier=spam adjust=0 reason=mlx scancount=1 engine=8.12.0-2103310000
 definitions=main-2104020008
X-Proofpoint-GUID: 7V4m2zuqMf4Z2s3fZfv1AhJ3CokvQ8PL
X-Proofpoint-ORIG-GUID: 7V4m2zuqMf4Z2s3fZfv1AhJ3CokvQ8PL
X-Proofpoint-Virus-Version: vendor=nai engine=6200 definitions=9941
 signatures=668683
X-Proofpoint-Spam-Details: rule=notspam policy=default score=0 spamscore=0
 mlxlogscore=999 mlxscore=0
 lowpriorityscore=0 suspectscore=0 priorityscore=1501 phishscore=0
 clxscore=1015 impostorscore=0 malwarescore=0 bulkscore=0 adultscore=0
 classifier=spam adjust=0 reason=mlx scancount=1 engine=8.12.0-2103310000
 definitions=main-2104020008
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

Define the actual size of the IOPM and MSRPM tables so that the actual size
can be used when initializing them and when checking the consistency of the
physical addresses. These #defines are placed in svm.h so that they can be
shared.

Suggested-by: Sean Christopherson <seanjc@google.com>
Signed-off-by: Krish Sadhukhan <krish.sadhukhan@oracle.com>
---
 arch/x86/kvm/svm/svm.c | 20 ++++++++++----------
 arch/x86/kvm/svm/svm.h |  3 +++
 2 files changed, 13 insertions(+), 10 deletions(-)

diff --git a/arch/x86/kvm/svm/svm.c b/arch/x86/kvm/svm/svm.c
index 58a45bb139f8..d1dd6539ed00 100644
--- a/arch/x86/kvm/svm/svm.c
+++ b/arch/x86/kvm/svm/svm.c
@@ -56,9 +56,6 @@ static const struct x86_cpu_id svm_cpu_id[] = {
 MODULE_DEVICE_TABLE(x86cpu, svm_cpu_id);
 #endif
 
-#define IOPM_ALLOC_ORDER 2
-#define MSRPM_ALLOC_ORDER 1
-
 #define SEG_TYPE_LDT 2
 #define SEG_TYPE_BUSY_TSS16 3
 
@@ -681,14 +678,15 @@ void set_msr_interception(struct kvm_vcpu *vcpu, u32 *msrpm, u32 msr,
 
 u32 *svm_vcpu_alloc_msrpm(void)
 {
-	struct page *pages = alloc_pages(GFP_KERNEL_ACCOUNT, MSRPM_ALLOC_ORDER);
+	unsigned int order = get_order(MSRPM_ALLOC_SIZE);
+	struct page *pages = alloc_pages(GFP_KERNEL_ACCOUNT, order);
 	u32 *msrpm;
 
 	if (!pages)
 		return NULL;
 
 	msrpm = page_address(pages);
-	memset(msrpm, 0xff, PAGE_SIZE * (1 << MSRPM_ALLOC_ORDER));
+	memset(msrpm, 0xff, PAGE_SIZE * (1 << order));
 
 	return msrpm;
 }
@@ -707,7 +705,7 @@ void svm_vcpu_init_msrpm(struct kvm_vcpu *vcpu, u32 *msrpm)
 
 void svm_vcpu_free_msrpm(u32 *msrpm)
 {
-	__free_pages(virt_to_page(msrpm), MSRPM_ALLOC_ORDER);
+	__free_pages(virt_to_page(msrpm), get_order(MSRPM_ALLOC_SIZE));
 }
 
 static void svm_msr_filter_changed(struct kvm_vcpu *vcpu)
@@ -894,7 +892,8 @@ static void svm_hardware_teardown(void)
 	for_each_possible_cpu(cpu)
 		svm_cpu_uninit(cpu);
 
-	__free_pages(pfn_to_page(iopm_base >> PAGE_SHIFT), IOPM_ALLOC_ORDER);
+	__free_pages(pfn_to_page(iopm_base >> PAGE_SHIFT),
+	    get_order(IOPM_ALLOC_SIZE));
 	iopm_base = 0;
 }
 
@@ -930,14 +929,15 @@ static __init int svm_hardware_setup(void)
 	struct page *iopm_pages;
 	void *iopm_va;
 	int r;
+	unsigned int order = get_order(IOPM_ALLOC_SIZE);
 
-	iopm_pages = alloc_pages(GFP_KERNEL, IOPM_ALLOC_ORDER);
+	iopm_pages = alloc_pages(GFP_KERNEL, order);
 
 	if (!iopm_pages)
 		return -ENOMEM;
 
 	iopm_va = page_address(iopm_pages);
-	memset(iopm_va, 0xff, PAGE_SIZE * (1 << IOPM_ALLOC_ORDER));
+	memset(iopm_va, 0xff, PAGE_SIZE * (1 << order));
 	iopm_base = page_to_pfn(iopm_pages) << PAGE_SHIFT;
 
 	init_msrpm_offsets();
@@ -1408,7 +1408,7 @@ static void svm_free_vcpu(struct kvm_vcpu *vcpu)
 	sev_free_vcpu(vcpu);
 
 	__free_page(pfn_to_page(__sme_clr(svm->vmcb_pa) >> PAGE_SHIFT));
-	__free_pages(virt_to_page(svm->msrpm), MSRPM_ALLOC_ORDER);
+	__free_pages(virt_to_page(svm->msrpm), get_order(MSRPM_ALLOC_SIZE));
 }
 
 static void svm_prepare_guest_switch(struct kvm_vcpu *vcpu)
diff --git a/arch/x86/kvm/svm/svm.h b/arch/x86/kvm/svm/svm.h
index 39e071fdab0c..d0a4d7ce8445 100644
--- a/arch/x86/kvm/svm/svm.h
+++ b/arch/x86/kvm/svm/svm.h
@@ -28,6 +28,9 @@ static const u32 host_save_user_msrs[] = {
 };
 #define NR_HOST_SAVE_USER_MSRS ARRAY_SIZE(host_save_user_msrs)
 
+#define IOPM_ALLOC_SIZE PAGE_SIZE * 3
+#define MSRPM_ALLOC_SIZE PAGE_SIZE * 2
+
 #define MAX_DIRECT_ACCESS_MSRS	18
 #define MSRPM_OFFSETS	16
 extern u32 msrpm_offsets[MSRPM_OFFSETS] __read_mostly;

From patchwork Fri Apr  2 00:43:28 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Krish Sadhukhan <krish.sadhukhan@oracle.com>
X-Patchwork-Id: 12180175
Return-Path: <kvm-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-15.9 required=3.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,HEADER_FROM_DIFFERENT_DOMAINS,INCLUDES_CR_TRAILER,
	INCLUDES_PATCH,MAILING_LIST_MULTI,MSGID_FROM_MTA_HEADER,SPF_HELO_NONE,
	SPF_PASS,UNWANTED_LANGUAGE_BODY,USER_AGENT_GIT autolearn=ham
	autolearn_force=no version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id 010E9C433B4
	for <kvm@archiver.kernel.org>; Fri,  2 Apr 2021 01:33:05 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id C4E8060241
	for <kvm@archiver.kernel.org>; Fri,  2 Apr 2021 01:33:05 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S234043AbhDBBdC (ORCPT <rfc822;kvm@archiver.kernel.org>);
        Thu, 1 Apr 2021 21:33:02 -0400
Received: from aserp2120.oracle.com ([141.146.126.78]:33978 "EHLO
        aserp2120.oracle.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S233665AbhDBBdA (ORCPT <rfc822;kvm@vger.kernel.org>);
        Thu, 1 Apr 2021 21:33:00 -0400
Received: from pps.filterd (aserp2120.oracle.com [127.0.0.1])
        by aserp2120.oracle.com (8.16.0.42/8.16.0.42) with SMTP id
 1321W1kZ009888;
        Fri, 2 Apr 2021 01:32:55 GMT
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=oracle.com;
 h=from : to : cc :
 subject : date : message-id : in-reply-to : references :
 content-transfer-encoding : content-type : mime-version;
 s=corp-2020-01-29; bh=loDUW8sWl+KyDRD60wmhRxXhecJPMSrqFNBuZsnGxXk=;
 b=Xj9EoPbPw6JZvUCE4LeeDI6aJMblteSBZrIh6wk7AP2EhGtoiK138l5/H2fv2z0XfPD+
 /D9kE6HmPH5IzxODI6o5eGE2g6F/UKfbKP8tSRoPImsbclCRd057dpxv7BQBeOGnQhgg
 6AdGgaxgsXGiCpBRx7Oc7jR+84wdT8hFAO/gE95inwB2/W+7PJmV0oeh8LJCAKbMQEv3
 StRiuOfdeaDxh2PtXCgLuVrXWmzhc9rTQ+pFKW0W+8bL8iIQa+z18RK8mum5l7FzJMsm
 9gmk9oKMKjMTIwNh+hvD7MY5WuDLsoktepxIrurtk+qWZbrc57OjlErKCai4M2bkKwTe QQ==
Received: from aserp3030.oracle.com (aserp3030.oracle.com [141.146.126.71])
        by aserp2120.oracle.com with ESMTP id 37n2akkk5x-1
        (version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256
 verify=OK);
        Fri, 02 Apr 2021 01:32:55 +0000
Received: from pps.filterd (aserp3030.oracle.com [127.0.0.1])
        by aserp3030.oracle.com (8.16.0.42/8.16.0.42) with SMTP id
 1321TkIq150731;
        Fri, 2 Apr 2021 01:32:55 GMT
Received: from nam10-mw2-obe.outbound.protection.outlook.com
 (mail-mw2nam10lp2101.outbound.protection.outlook.com [104.47.55.101])
        by aserp3030.oracle.com with ESMTP id 37n2at0196-1
        (version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256
 verify=OK);
        Fri, 02 Apr 2021 01:32:55 +0000
ARC-Seal: i=1; a=rsa-sha256; s=arcselector9901; d=microsoft.com; cv=none;
 b=a/KEqFFVz3OnOjJRABsqAx4Ndlww0TZ7bhkLIdDwvvLANCAgzyr3vsJpxTTJMZcc4nOYDkifJZSNrZk01eZjS2U0C+zd+hMdozfTepjI7neH3+Abf/7BoAAzGkUEWbhmYPYuEUwsreNcGGCdq0DFjIve06i5HsM/2AaDDdkpAbaspWfyRxFjKwUEizN+B8uVpSn7zXynK5kRqjJA00HsRuId43NSX84Vve2Si4Y/XuhGITauFLoL1oABtUYronmVRvd832lmxaEFyU7MCOPwkQoUmIgyJ5oQQNh0EK6yE++2a1HNlGGmLBWX3LZi1dI1lJu14nh2fB2lXq1tPJQd4w==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=microsoft.com;
 s=arcselector9901;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=loDUW8sWl+KyDRD60wmhRxXhecJPMSrqFNBuZsnGxXk=;
 b=Ir31EPt+klErjeLYLNwig2zt/JGdDW4Z84Y0JgJV2I5C1RcO03FiqaOXMrk0D1iDSPS3hguBIjro9Ab7PR2B1stBTenTIMwDZ/DQqbvN5HAeiLdatK4/7PLk9PI7yMlFVujH+XjG1FI51on6BXEmk4asP3JqcOk7QKQUOWV30c66CEXeccawCgFXVEfjHZ7DRtMnoHGy0r1WisANuERVMAY0Ai6yZbfdBSDLymx+8AQEASdpWwwQqQ4vh9wnLMDB1uQwszye7AxAQklLmADCzi3AXqRCf6vkgiJ9/eMfFZO4R2Ft5eiI7spxlQCAgOsBMmKVrHLGVZxMv3ks2INefA==
ARC-Authentication-Results: i=1; mx.microsoft.com 1; spf=pass
 smtp.mailfrom=oracle.com; dmarc=pass action=none header.from=oracle.com;
 dkim=pass header.d=oracle.com; arc=none
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=oracle.onmicrosoft.com; s=selector2-oracle-onmicrosoft-com;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=loDUW8sWl+KyDRD60wmhRxXhecJPMSrqFNBuZsnGxXk=;
 b=sA/ospFP2VeNXfyoM8Bfjdorvl0jympQgx7PAJe51JQVDUdsnGrJv5npV75B2d/635TkoF9GgiTCPOCJelyzC/zsIM2+a5ksLabrAKJmI5vcIPWRQFtgGqwG3x26X4DpQ4ViGstYyBdkH6pdCxYGbvIRPE0W1jh5nvXx0i/E1mo=
Authentication-Results: vger.kernel.org; dkim=none (message not signed)
 header.d=none;vger.kernel.org; dmarc=none action=none header.from=oracle.com;
Received: from SN6PR10MB3021.namprd10.prod.outlook.com (2603:10b6:805:cc::19)
 by SA2PR10MB4795.namprd10.prod.outlook.com (2603:10b6:806:11d::14) with
 Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.3999.28; Fri, 2 Apr
 2021 01:32:53 +0000
Received: from SN6PR10MB3021.namprd10.prod.outlook.com
 ([fe80::5911:9489:e05c:2d44]) by SN6PR10MB3021.namprd10.prod.outlook.com
 ([fe80::5911:9489:e05c:2d44%5]) with mapi id 15.20.3999.029; Fri, 2 Apr 2021
 01:32:53 +0000
From: Krish Sadhukhan <krish.sadhukhan@oracle.com>
To: kvm@vger.kernel.org
Cc: pbonzini@redhat.com, jmattson@google.com, seanjc@google.com
Subject: [PATCH 2/5 v6] nSVM: Check addresses of MSR and IO permission maps
Date: Thu,  1 Apr 2021 20:43:28 -0400
Message-Id: <20210402004331.91658-3-krish.sadhukhan@oracle.com>
X-Mailer: git-send-email 2.25.4
In-Reply-To: <20210402004331.91658-1-krish.sadhukhan@oracle.com>
References: <20210402004331.91658-1-krish.sadhukhan@oracle.com>
X-Originating-IP: [138.3.201.29]
X-ClientProxiedBy: SJ0PR13CA0024.namprd13.prod.outlook.com
 (2603:10b6:a03:2c0::29) To SN6PR10MB3021.namprd10.prod.outlook.com
 (2603:10b6:805:cc::19)
MIME-Version: 1.0
X-MS-Exchange-MessageSentRepresentingType: 1
Received: from ban25x6uut29.us.oracle.com (138.3.201.29) by
 SJ0PR13CA0024.namprd13.prod.outlook.com (2603:10b6:a03:2c0::29) with
 Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.4020.9 via Frontend
 Transport; Fri, 2 Apr 2021 01:32:52 +0000
X-MS-PublicTrafficType: Email
X-MS-Office365-Filtering-Correlation-Id: 4642ec84-5066-42fc-d929-08d8f5773c20
X-MS-TrafficTypeDiagnostic: SA2PR10MB4795:
X-Microsoft-Antispam-PRVS: 
 <SA2PR10MB4795B9CA64DC724643E103D4817A9@SA2PR10MB4795.namprd10.prod.outlook.com>
X-MS-Oob-TLC-OOBClassifiers: OLM:5236;
X-MS-Exchange-SenderADCheck: 1
X-Microsoft-Antispam: BCL:0;
X-Microsoft-Antispam-Message-Info: 
 Ilb7V8oDi3rlTm9HCCbAJ0qTjhXXp1QZ275apehPANJPEqPhMPJt8Msy5cZN9aO7Gkkhy/hWX225nOo8vaK5bAelkS9hCYl96AhytWE2FQ6OUcKiwswPyx6/QbhNSif1hsjEiOw8bXQwjhS4taTe9O1SOovHBnIhkJnj5YKK9XAVAWtsZThKFzsDuI21m/06Rez+04L4Lpp/fR223sXpHIo59vtzGqLXnv17TibRw913gQs2R3VxYeBFtYIN2DLxX7qhTuuF5l1IhR8vBMR/aa87k1VUvEMGTBGfISP7De6LvRQZBeZd1N9SWKbAe9z2NmM+3IYIFe+b1ej3VN9vWVKKrudc9WXWECp7/YwTrv3mTLqxc53WyX50Bvg0VMApteGIxxFhavliUrVQiv41Wnub1quRXsWSqlzC0MJg0O8jY3hk8QEnAbu45IEXeAMWVMSZUq4SMkCciBsCAJLTNS4DE9mraD9MIzazk+N8AEJcZiWuuu/Qf3Lzc/1DGOIUWF/Nzf4b+7bxo7jhyVIT0ZaEURS91J5e0SEUWPCA/WXRiuCKaGw866bUXIPrhoDl/UyaF9Kav7WwhzXCAw+nhHzgH3rHZBIjjb3G1PgKdKwxW12QZIpfABJx8+xBMAEhKyNUnuS7xFodyghPQp/qfA==
X-Forefront-Antispam-Report: 
 CIP:255.255.255.255;CTRY:;LANG:en;SCL:1;SRV:;IPV:NLI;SFV:NSPM;H:SN6PR10MB3021.namprd10.prod.outlook.com;PTR:;CAT:NONE;SFS:(376002)(366004)(346002)(396003)(39860400002)(136003)(6666004)(26005)(36756003)(316002)(83380400001)(1076003)(66556008)(5660300002)(16526019)(66476007)(186003)(7696005)(6486002)(2616005)(44832011)(52116002)(956004)(8936002)(38100700001)(8676002)(6916009)(2906002)(66946007)(86362001)(4326008)(478600001);DIR:OUT;SFP:1101;
X-MS-Exchange-AntiSpam-MessageData: 
 JJR6fw5ef+pwDK2O5D4V44FggQRdiDn4TIxb9zWDifusCjCRrRIpKrpWC7PM1a7NXhEmVAdbNLsISgAVsozuvqXnJS/1Y/va0Z9HpX54oqxN7sngBoLNPyuuCwHILjN+Xk/IUrc7loFZAgIR7IZ/RKBF4YkMjCtFzf6n6gXXfr0XCX4gXHspsa8i9U0EMyuUoQ6tzdEItHGmW9DP0C5wVDYHfZsvhlwTLn8pdVuEQ9vWUqei2lYY/wIolG/GqeFMrQri7uwKx16SbkxAKi00ajXvXN0iwFfrkBdpDB0sPgjOxqzTyN9GTUDKL3DL2696BOFyGq0Th9tzwxbcQ22E5hw68QDzEDtWVZup+14Naa/Q2f9HDyfNjtjMkENPdCFV8c+lDtFiJuCNhDWMNqatzc5myeb9JFWdrTM9H88op5vrrJul5KvrjCMQPtHMmvePPuHqyEChpdmA6p7NQ4lVNsSUAgQyaqredQeZPTuIZIGnBV/JTAWVTBLL1GLDoz3wBk4FDEn2teLyQmiizRKYIHkV+Tid9NKo/4cncJz5vlG+rg4s27zuFTXLZi7xVb2974SCr6cDDpteCURjLzxE/D3otXyhDA1/bA5Cb0B1c2z5F3KQIzzSndeOOBMuEYOH8YrBqAxpOlFGG4XJpNmbgB2DWKrVWiRvWbe3HQvK6yAYfEkVchP5JBE/KTSZcn5qDfI3DXLR3c85uKxym479WQMcvtNOAr+l5tTXBhWOj67v2sfP6S2qHX1BPp0P55d/2d2hNmsFRUriORU2CufUdk7zKiZPL6lGmfMhR1kpcYTM+aBhX3lKV5405Vy6Tsi499S9ZYndA03NFIE/Ky0BcQ+nvL+rk+GCtCaiqALfnaBIjCsDzNReORVc3eWrvQeQnPgNgCy9xL6WixZK6wvKIw/OurvdBmMlEDB5hg0i01kYpMAKYKa3m14aSdlTCgEm0AFrvowQJgXcWIlTj1lCj18CJUEntrSNQF6RLit6GWWin87dIQCvCsays5FGQwyBn8h+nQ7Hd9/sIxRwxNi+zb2W5LrhKFK8abJPS9xoRj3lTgeDe33klY6lXTuxRSn2pJZitjm18qXN1onwZQTKzWn7qd8DjM+GWoTtI5jE/FXHEcETcsCaPYpYCmrEMRycS6sOlHWdfFL4WD/C2Fe+56syj2+Q3axsStBeK+L1alQmjO2rv2xwyR2iLh/tlOzN+J+TuX2mfoq7nF0gE6kEK8M/qbXmaXJKbt4ADbCsyrtuBz7+BW0FiRNoqRQzEfdJZj3JouSkYedAWj/Dg7QZJxxwIYxNzUtFj00moDK08GJgkKTbsI+Yz5+4lcLKUVeO
X-OriginatorOrg: oracle.com
X-MS-Exchange-CrossTenant-Network-Message-Id: 
 4642ec84-5066-42fc-d929-08d8f5773c20
X-MS-Exchange-CrossTenant-AuthSource: SN6PR10MB3021.namprd10.prod.outlook.com
X-MS-Exchange-CrossTenant-AuthAs: Internal
X-MS-Exchange-CrossTenant-OriginalArrivalTime: 02 Apr 2021 01:32:53.3802
 (UTC)
X-MS-Exchange-CrossTenant-FromEntityHeader: Hosted
X-MS-Exchange-CrossTenant-Id: 4e2c6054-71cb-48f1-bd6c-3a9705aca71b
X-MS-Exchange-CrossTenant-MailboxType: HOSTED
X-MS-Exchange-CrossTenant-UserPrincipalName: 
 mbch2/RbA7oBas52FwfaPooLgSvZsdOCNOaMVu8UH+diWwvXZjElibYygqnOs/MZpAMbu7oGgyKnluYUVMGOHU2cw2lHnzSO7RNNZRetmx4=
X-MS-Exchange-Transport-CrossTenantHeadersStamped: SA2PR10MB4795
X-Proofpoint-Virus-Version: vendor=nai engine=6200 definitions=9941
 signatures=668683
X-Proofpoint-Spam-Details: rule=notspam policy=default score=0 mlxlogscore=999
 spamscore=0
 suspectscore=0 bulkscore=0 mlxscore=0 adultscore=0 malwarescore=0
 phishscore=0 classifier=spam adjust=0 reason=mlx scancount=1
 engine=8.12.0-2103310000 definitions=main-2104020008
X-Proofpoint-ORIG-GUID: _s8v65JdVeTJxjrrD4jeV894zAPmEEpa
X-Proofpoint-GUID: _s8v65JdVeTJxjrrD4jeV894zAPmEEpa
X-Proofpoint-Virus-Version: vendor=nai engine=6200 definitions=9941
 signatures=668683
X-Proofpoint-Spam-Details: rule=notspam policy=default score=0 mlxscore=0
 impostorscore=0 phishscore=0
 bulkscore=0 adultscore=0 clxscore=1015 malwarescore=0 priorityscore=1501
 suspectscore=0 spamscore=0 mlxlogscore=999 lowpriorityscore=0
 classifier=spam adjust=0 reason=mlx scancount=1 engine=8.12.0-2103310000
 definitions=main-2104020008
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

According to section "Canonicalization and Consistency Checks" in APM vol 2,
the following guest state is illegal:

    "The MSR or IOIO intercept tables extend to a physical address that
     is greater than or equal to the maximum supported physical address."

Suggested-by: Paolo Bonzini <pbonzini@redhat.com>
Signed-off-by: Krish Sadhukhan <krish.sadhukhan@oracle.com>
---
 arch/x86/kvm/svm/nested.c | 23 +++++++++++++++++++----
 1 file changed, 19 insertions(+), 4 deletions(-)

diff --git a/arch/x86/kvm/svm/nested.c b/arch/x86/kvm/svm/nested.c
index fb204eaa8bb3..b3988b3a3fd5 100644
--- a/arch/x86/kvm/svm/nested.c
+++ b/arch/x86/kvm/svm/nested.c
@@ -231,7 +231,15 @@ static bool svm_get_nested_state_pages(struct kvm_vcpu *vcpu)
 	return true;
 }
 
-static bool nested_vmcb_check_controls(struct vmcb_control_area *control)
+static bool nested_svm_check_bitmap_pa(struct kvm_vcpu *vcpu, u64 pa,
+				       u32 size)
+{
+	u64 last_pa = PAGE_ALIGN(pa) + size - 1;
+	return (kvm_vcpu_is_legal_gpa(vcpu, last_pa));
+}
+
+static bool nested_vmcb_check_controls(struct kvm_vcpu *vcpu,
+				       struct vmcb_control_area *control)
 {
 	if ((vmcb_is_intercept(control, INTERCEPT_VMRUN)) == 0)
 		return false;
@@ -243,6 +251,13 @@ static bool nested_vmcb_check_controls(struct vmcb_control_area *control)
 	    !npt_enabled)
 		return false;
 
+	if (!nested_svm_check_bitmap_pa(vcpu, control->msrpm_base_pa,
+	    MSRPM_ALLOC_SIZE))
+		return false;
+	if (!nested_svm_check_bitmap_pa(vcpu, control->iopm_base_pa,
+	    IOPM_ALLOC_SIZE - PAGE_SIZE + 1))
+		return false;
+
 	return true;
 }
 
@@ -275,7 +290,7 @@ static bool nested_vmcb_check_save(struct vcpu_svm *svm, struct vmcb *vmcb12)
 		    kvm_vcpu_is_illegal_gpa(vcpu, vmcb12->save.cr3))
 			return false;
 	}
-	if (!kvm_is_valid_cr4(&svm->vcpu, vmcb12->save.cr4))
+	if (!kvm_is_valid_cr4(vcpu, vmcb12->save.cr4))
 		return false;
 
 	return true;
@@ -531,7 +546,7 @@ int nested_svm_vmrun(struct vcpu_svm *svm)
 	load_nested_vmcb_control(svm, &vmcb12->control);
 
 	if (!nested_vmcb_check_save(svm, vmcb12) ||
-	    !nested_vmcb_check_controls(&svm->nested.ctl)) {
+	    !nested_vmcb_check_controls(&svm->vcpu, &svm->nested.ctl)) {
 		vmcb12->control.exit_code    = SVM_EXIT_ERR;
 		vmcb12->control.exit_code_hi = 0;
 		vmcb12->control.exit_info_1  = 0;
@@ -1207,7 +1222,7 @@ static int svm_set_nested_state(struct kvm_vcpu *vcpu,
 		goto out_free;
 
 	ret = -EINVAL;
-	if (!nested_vmcb_check_controls(ctl))
+	if (!nested_vmcb_check_controls(vcpu, ctl))
 		goto out_free;
 
 	/*

From patchwork Fri Apr  2 00:43:29 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Krish Sadhukhan <krish.sadhukhan@oracle.com>
X-Patchwork-Id: 12180173
Return-Path: <kvm-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-18.8 required=3.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,HEADER_FROM_DIFFERENT_DOMAINS,INCLUDES_CR_TRAILER,
	INCLUDES_PATCH,MAILING_LIST_MULTI,MSGID_FROM_MTA_HEADER,SPF_HELO_NONE,
	SPF_PASS,USER_AGENT_GIT autolearn=ham autolearn_force=no version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id EDFDDC433ED
	for <kvm@archiver.kernel.org>; Fri,  2 Apr 2021 01:33:05 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id A982160FF3
	for <kvm@archiver.kernel.org>; Fri,  2 Apr 2021 01:33:05 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S234015AbhDBBdB (ORCPT <rfc822;kvm@archiver.kernel.org>);
        Thu, 1 Apr 2021 21:33:01 -0400
Received: from aserp2130.oracle.com ([141.146.126.79]:49852 "EHLO
        aserp2130.oracle.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S231179AbhDBBdA (ORCPT <rfc822;kvm@vger.kernel.org>);
        Thu, 1 Apr 2021 21:33:00 -0400
Received: from pps.filterd (aserp2130.oracle.com [127.0.0.1])
        by aserp2130.oracle.com (8.16.0.42/8.16.0.42) with SMTP id
 1321Wr5n063887;
        Fri, 2 Apr 2021 01:32:56 GMT
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=oracle.com;
 h=from : to : cc :
 subject : date : message-id : in-reply-to : references :
 content-transfer-encoding : content-type : mime-version;
 s=corp-2020-01-29; bh=YwdOVBRL4lgF1P+4RwcjwaVT/DZRjxi78ki3Vg3kac0=;
 b=YLConeR0OivaBqrmhf5gj0uBklxpJ95kgF2tpHeaO1KcXHDcLsE+hA6WCaDeiKwL3JNN
 dUzVDDwI7ZCuQZGqoJvRNYgbD+8qS2l7ATXpTYIxeW9fdm000zJT0qinGrp5aZDaM3G6
 A4EfPRw4/aoJ6cTzkRE0ef5axogK+nm8LiPjZdVzmutjolDagaioAUh9luqDJEkqVZX1
 7cJJUJOQ0KDzi41hu7MEcUxwvw2RAb5ZRdk1ZH6AA03igHY19WB5zbwCCDe9Gio9UhID
 nYRDlP6YUJRXfwykKDo+KnmQS1zrGUQgfQ22o3RqrNuxFqL5THB850srxo1ijmqgtLvp TA==
Received: from aserp3030.oracle.com (aserp3030.oracle.com [141.146.126.71])
        by aserp2130.oracle.com with ESMTP id 37n33dugm4-1
        (version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256
 verify=OK);
        Fri, 02 Apr 2021 01:32:56 +0000
Received: from pps.filterd (aserp3030.oracle.com [127.0.0.1])
        by aserp3030.oracle.com (8.16.0.42/8.16.0.42) with SMTP id
 1321TkIr150731;
        Fri, 2 Apr 2021 01:32:56 GMT
Received: from nam10-mw2-obe.outbound.protection.outlook.com
 (mail-mw2nam10lp2101.outbound.protection.outlook.com [104.47.55.101])
        by aserp3030.oracle.com with ESMTP id 37n2at0196-2
        (version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256
 verify=OK);
        Fri, 02 Apr 2021 01:32:55 +0000
ARC-Seal: i=1; a=rsa-sha256; s=arcselector9901; d=microsoft.com; cv=none;
 b=NuYOqnlPTmJ7ks71M+QnbqB7ScJjNyyPAuPWPMYbwbZEI4bvRlIh//xWd+FBhsRkmIEMKbTTC4ca8MvC088eqOjaobS0G2/Mb/zak7uiA4fWpLt25X3laQvvg5mpbZRiKJmPj+MQK77DPxCPO3Ep8n2xq8NibxH+f7+scLSvgPVkHxvsAHxZBaP84jRWk2A2mRKxzm3ulc1DXuueuDS78rKjHIVDGun8sFQpdrI5gJ/cD629cHa4meAG2uwYZobLLiWcWaRPpoT/Mv5W4MDDAoZ2GxsaSxaWj1CtRlsvXO6MK9AO+Zhn+GZPyphlRQMQusFlnsKhstaKRaUpL3wiJQ==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=microsoft.com;
 s=arcselector9901;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=YwdOVBRL4lgF1P+4RwcjwaVT/DZRjxi78ki3Vg3kac0=;
 b=Qil/jGm4jOQmTtR4as3XFOtWUU6TTAumUWZT9b/2dnEnEclDfMbipazTB7Yq7ZRp4d1i+uOfnE0yVsvAs1wsOQl4Ts/WqRbKNTIl5CcDQ+aZtxHGxh/KHwoxQBBb8RBILclJik/IeQicU6jm1OjX95wVTHyRAQjY/t2KDUrkJq/fC9bwCphVrbx2RealsyPymMMDUW5v8bs7gT9yEzg/GX1ytQgI/DUMZKSXsnmvl0II+aG4HbPjVjdMDyHKa76YaMbu2xItympE/9UGYR669s+ZlZeYWqBqk/Xvh4eTPYtR13Pbv804DZrTzgd1iUipjD533nCL5jZUES32hkiGTA==
ARC-Authentication-Results: i=1; mx.microsoft.com 1; spf=pass
 smtp.mailfrom=oracle.com; dmarc=pass action=none header.from=oracle.com;
 dkim=pass header.d=oracle.com; arc=none
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=oracle.onmicrosoft.com; s=selector2-oracle-onmicrosoft-com;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=YwdOVBRL4lgF1P+4RwcjwaVT/DZRjxi78ki3Vg3kac0=;
 b=mW1S92JMc5ltYQkMNuWzCNuhWP97Atxmu9tAUn81pj8v/3ieQ4dtxp05lMMr0rhjHKKPkPNXSizot2k5sO5i7vx4IME2f3jK1z+PUXS5rLDVjVHiKcM+8mnSzzYyq7AOMiTRRgQ89GuJI3IqYbaEu60OmeGxcVGT3wOhuPkmHgU=
Authentication-Results: vger.kernel.org; dkim=none (message not signed)
 header.d=none;vger.kernel.org; dmarc=none action=none header.from=oracle.com;
Received: from SN6PR10MB3021.namprd10.prod.outlook.com (2603:10b6:805:cc::19)
 by SA2PR10MB4795.namprd10.prod.outlook.com (2603:10b6:806:11d::14) with
 Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.3999.28; Fri, 2 Apr
 2021 01:32:54 +0000
Received: from SN6PR10MB3021.namprd10.prod.outlook.com
 ([fe80::5911:9489:e05c:2d44]) by SN6PR10MB3021.namprd10.prod.outlook.com
 ([fe80::5911:9489:e05c:2d44%5]) with mapi id 15.20.3999.029; Fri, 2 Apr 2021
 01:32:54 +0000
From: Krish Sadhukhan <krish.sadhukhan@oracle.com>
To: kvm@vger.kernel.org
Cc: pbonzini@redhat.com, jmattson@google.com, seanjc@google.com
Subject: [PATCH 3/5 v6] KVM: nSVM: Cleanup in nested_svm_vmrun()
Date: Thu,  1 Apr 2021 20:43:29 -0400
Message-Id: <20210402004331.91658-4-krish.sadhukhan@oracle.com>
X-Mailer: git-send-email 2.25.4
In-Reply-To: <20210402004331.91658-1-krish.sadhukhan@oracle.com>
References: <20210402004331.91658-1-krish.sadhukhan@oracle.com>
X-Originating-IP: [138.3.201.29]
X-ClientProxiedBy: SJ0PR13CA0024.namprd13.prod.outlook.com
 (2603:10b6:a03:2c0::29) To SN6PR10MB3021.namprd10.prod.outlook.com
 (2603:10b6:805:cc::19)
MIME-Version: 1.0
X-MS-Exchange-MessageSentRepresentingType: 1
Received: from ban25x6uut29.us.oracle.com (138.3.201.29) by
 SJ0PR13CA0024.namprd13.prod.outlook.com (2603:10b6:a03:2c0::29) with
 Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.4020.9 via Frontend
 Transport; Fri, 2 Apr 2021 01:32:53 +0000
X-MS-PublicTrafficType: Email
X-MS-Office365-Filtering-Correlation-Id: 3c757095-0814-4c5b-9b1d-08d8f5773cea
X-MS-TrafficTypeDiagnostic: SA2PR10MB4795:
X-Microsoft-Antispam-PRVS: 
 <SA2PR10MB47959AFDE18043E1309435B2817A9@SA2PR10MB4795.namprd10.prod.outlook.com>
X-MS-Oob-TLC-OOBClassifiers: OLM:4714;
X-MS-Exchange-SenderADCheck: 1
X-Microsoft-Antispam: BCL:0;
X-Microsoft-Antispam-Message-Info: 
 h6SkGRexuOf2S/Dx57S3uiz+t+AGubtfodCscQ/4iBAYjaXmBizK5OJ2HIqItilrYTx2UfB6hVF1e+UuoCZz9ryeBiqm5HlK2XHovMwmDDEKF+1EWHkPjqR0rDnd2URDlG7lyMoWQwsJCFu5aV0kSqRMu5i2XR5zNs1s774P6qih7uXwr9y8lmXHxj/cUhlaqRfjYJoZYR0RfUtIko1W+KUmqWfsV1i+jV10xaO+PthdGYNL3ZBSm7WrWg9p9fnE2jzTfdTwEOTfZaKWKz7yE0Uxuc8Bx2mQzlYyjYpkBtc9OrCJGJCx9x2mUry7/n75toSCGyU4ksPookRAOf8yk16UffmVdPEDgPQx8/Dq91IvNWoghOc4x/lI9ZNnm3wgNY3cu1nkeBPPQoNdagZy/QXYpadycgIlnj3SKZiCuNfZV/Y9wkHrRBSFRWdptn0dTBGHsRW/yOK8jyimQ0NHBsCfkFB/SGThzUpe2EUnva6OHj6xpf+TE315AWOuhALPoOl4HGk0Ts718hZyBICG8ykpOwFNub5yMPhfqcU5fYyif9kJIpp8QydgYFcRNyrl1xnE1AS9IAH7nWM5oy2xXbHRBnnXgG06FENYfJZ3MCKpRzDTtm1kyoTHN5WEBm04VvQ9rD6hBLvjzPXSNNkAVg==
X-Forefront-Antispam-Report: 
 CIP:255.255.255.255;CTRY:;LANG:en;SCL:1;SRV:;IPV:NLI;SFV:NSPM;H:SN6PR10MB3021.namprd10.prod.outlook.com;PTR:;CAT:NONE;SFS:(376002)(366004)(346002)(396003)(39860400002)(136003)(6666004)(26005)(36756003)(316002)(83380400001)(1076003)(66556008)(5660300002)(16526019)(66476007)(186003)(7696005)(6486002)(2616005)(44832011)(52116002)(956004)(8936002)(38100700001)(8676002)(6916009)(2906002)(66946007)(86362001)(4326008)(478600001);DIR:OUT;SFP:1101;
X-MS-Exchange-AntiSpam-MessageData: 
 CI8zCYv7sK7vs5H69N0sBtHkpaWrtvyj4s0exk/7nrBdmCQgRK76/FQd+mxAZUDoVVmTSYqgK9nR1+jPBlqEVbJvwhgItTryKHqTRIRLh+DY2EBYMij6gqzLMX0+FoPSqXhswEBmup/b0jCaS7m7ZYvsrGkqEYBrgZos0ML6r89MpSQFiJadjHiKCptDQpngDm4LADHhDPX/CvLKzCsAZjmNkYQESipSUYR0E3vEd1mXWdxf4BJotLPGSnK7szbqlTY8YyM/Jk3qb1aCyjyLtMbPkFPMQ8JnUl4d+xWTB4nkdLNY1wK9Am0NwA6DG/pV59efncu1q9i0lpbbX2OacvO9npk7RHpT4m1Yy7jTUXcZblHnsOoL/hIUjJmWl+q4a/euPQYadnvunO/U5/Q76sgVeFCZ3EfEGvBXk6nmN22a9rcRcpl+wkhGboRqp5udY0sM6AneyFNa37UAy3ztkZxUlmjSgrLBfxPA0cGfznudYRsptNKy7AZQt6TADkuj06oxft1AEhyQhIzhOHN2j90DiEcUGpn86yR8KW1J3jYcEHDc1nDs6IBOuQSlRgs0Qkkf+p3QnZfo0EtAVOR0pvuqzOONLCqIFXDuN05hNRWp9ZCbezc6IxhdpTOwB6AAgcsb/mJdhbJIAjriWyJXg/UYgrRPz8i3xpv2sR4cUOfN0fP4p4kvWqar9nZOEIy1pGSPmkQb4DpRmSdz77avm0pb/+9R+SiB4UflKQp6myrZ4MaD7NLGbbnkyxs+Y7g+OUJGviUPBPveVm+8bsaYcpFb3hJKlluHQdvs9CcSEBPWANpW9Qha+k94em2hez/7ppKbzwUQ/9TTVbjGR9WBX9ZNkaQO/vH84BYiADRQ+Jcb0bizvirY8B8i6y2ipWXwZwgJuXlDQBWGsljSxci95qwKFCTJwLJSl81ZGaK04O1cxL+vJpCKEe2m/vsHqRJg+KN2nhpt5d/27iyyMCu2v+Cv1xDiE+Hnu9jV8yeNjV1jVsz+JPNlrWfsOqU82/OCisn1c0av7dGJwsox1BQ6jewoNXnBBsh2/iMy0J9h6ciIv580e74G3HyNwow5wuqJKGLY9pE9Y2bgAQHf68IkhUc58gH/d8PaKKxqVq5M8yZeeOXAHldgF1uq8dxmCNJiCBjwL/zbTy364I5osLGja3PzZaAW7zxaiw8POHZ+4PFE+2bNYWqkmNwPDdC8VfvYY8291/jJiXPGBagEo75IEqgEXGMcO/HBiQqzkXEu2ggiAmbcQzZuMg9zj1pF77tuYNhEat7zR0iBNMkXyeDjdbc3PvII3JxfWmpImcbtTAc3LubcCjj8aSzaRgAq/rEE
X-OriginatorOrg: oracle.com
X-MS-Exchange-CrossTenant-Network-Message-Id: 
 3c757095-0814-4c5b-9b1d-08d8f5773cea
X-MS-Exchange-CrossTenant-AuthSource: SN6PR10MB3021.namprd10.prod.outlook.com
X-MS-Exchange-CrossTenant-AuthAs: Internal
X-MS-Exchange-CrossTenant-OriginalArrivalTime: 02 Apr 2021 01:32:54.8534
 (UTC)
X-MS-Exchange-CrossTenant-FromEntityHeader: Hosted
X-MS-Exchange-CrossTenant-Id: 4e2c6054-71cb-48f1-bd6c-3a9705aca71b
X-MS-Exchange-CrossTenant-MailboxType: HOSTED
X-MS-Exchange-CrossTenant-UserPrincipalName: 
 f73pWfkRD+UObPD3F9od/9qZ+HtQMyIaM0qTdtyj2A3ID3hHI55QLWwCaM1cBCSnR1IMHnhWt/j5uQ7xgaWOIu22Stxe0jRsOztH5CrhatM=
X-MS-Exchange-Transport-CrossTenantHeadersStamped: SA2PR10MB4795
X-Proofpoint-Virus-Version: vendor=nai engine=6200 definitions=9941
 signatures=668683
X-Proofpoint-Spam-Details: rule=notspam policy=default score=0 mlxlogscore=999
 spamscore=0
 suspectscore=0 bulkscore=0 mlxscore=0 adultscore=0 malwarescore=0
 phishscore=0 classifier=spam adjust=0 reason=mlx scancount=1
 engine=8.12.0-2103310000 definitions=main-2104020008
X-Proofpoint-GUID: q9RtLEYXVyTROYlN0m_HKyW0agIsGivg
X-Proofpoint-ORIG-GUID: q9RtLEYXVyTROYlN0m_HKyW0agIsGivg
X-Proofpoint-Virus-Version: vendor=nai engine=6200 definitions=9941
 signatures=668683
X-Proofpoint-Spam-Details: rule=notspam policy=default score=0 spamscore=0
 mlxlogscore=999 mlxscore=0
 lowpriorityscore=0 suspectscore=0 priorityscore=1501 phishscore=0
 clxscore=1015 impostorscore=0 malwarescore=0 bulkscore=0 adultscore=0
 classifier=spam adjust=0 reason=mlx scancount=1 engine=8.12.0-2103310000
 definitions=main-2104020008
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

Use local variables to derefence svm->vcpu and svm->vmcb as they make the
code tidier.

Suggested-by: Sean Christopherson <seanjc@google.com>
Signed-off-by: Krish Sadhukhan <krish.sadhukhan@oracle.com>
---
 arch/x86/kvm/svm/nested.c | 31 ++++++++++++++++---------------
 1 file changed, 16 insertions(+), 15 deletions(-)

diff --git a/arch/x86/kvm/svm/nested.c b/arch/x86/kvm/svm/nested.c
index b3988b3a3fd5..3fd51fe63170 100644
--- a/arch/x86/kvm/svm/nested.c
+++ b/arch/x86/kvm/svm/nested.c
@@ -517,26 +517,27 @@ int nested_svm_vmrun(struct vcpu_svm *svm)
 {
 	int ret;
 	struct vmcb *vmcb12;
+	struct kvm_vcpu *vcpu = &svm->vcpu;
 	struct vmcb *hsave = svm->nested.hsave;
 	struct vmcb *vmcb = svm->vmcb;
 	struct kvm_host_map map;
 	u64 vmcb12_gpa;
 
-	if (is_smm(&svm->vcpu)) {
-		kvm_queue_exception(&svm->vcpu, UD_VECTOR);
+	if (is_smm(vcpu)) {
+		kvm_queue_exception(vcpu, UD_VECTOR);
 		return 1;
 	}
 
-	vmcb12_gpa = svm->vmcb->save.rax;
-	ret = kvm_vcpu_map(&svm->vcpu, gpa_to_gfn(vmcb12_gpa), &map);
+	vmcb12_gpa = vmcb->save.rax;
+	ret = kvm_vcpu_map(vcpu, gpa_to_gfn(vmcb12_gpa), &map);
 	if (ret == -EINVAL) {
-		kvm_inject_gp(&svm->vcpu, 0);
+		kvm_inject_gp(vcpu, 0);
 		return 1;
 	} else if (ret) {
-		return kvm_skip_emulated_instruction(&svm->vcpu);
+		return kvm_skip_emulated_instruction(vcpu);
 	}
 
-	ret = kvm_skip_emulated_instruction(&svm->vcpu);
+	ret = kvm_skip_emulated_instruction(vcpu);
 
 	vmcb12 = map.hva;
 
@@ -556,8 +557,8 @@ int nested_svm_vmrun(struct vcpu_svm *svm)
 
 
 	/* Clear internal status */
-	kvm_clear_exception_queue(&svm->vcpu);
-	kvm_clear_interrupt_queue(&svm->vcpu);
+	kvm_clear_exception_queue(vcpu);
+	kvm_clear_interrupt_queue(vcpu);
 
 	/*
 	 * Save the old vmcb, so we don't need to pick what we save, but can
@@ -569,17 +570,17 @@ int nested_svm_vmrun(struct vcpu_svm *svm)
 	hsave->save.ds     = vmcb->save.ds;
 	hsave->save.gdtr   = vmcb->save.gdtr;
 	hsave->save.idtr   = vmcb->save.idtr;
-	hsave->save.efer   = svm->vcpu.arch.efer;
-	hsave->save.cr0    = kvm_read_cr0(&svm->vcpu);
+	hsave->save.efer   = vcpu->arch.efer;
+	hsave->save.cr0    = kvm_read_cr0(vcpu);
 	hsave->save.cr4    = svm->vcpu.arch.cr4;
-	hsave->save.rflags = kvm_get_rflags(&svm->vcpu);
-	hsave->save.rip    = kvm_rip_read(&svm->vcpu);
+	hsave->save.rflags = kvm_get_rflags(vcpu);
+	hsave->save.rip    = kvm_rip_read(vcpu);
 	hsave->save.rsp    = vmcb->save.rsp;
 	hsave->save.rax    = vmcb->save.rax;
 	if (npt_enabled)
 		hsave->save.cr3    = vmcb->save.cr3;
 	else
-		hsave->save.cr3    = kvm_read_cr3(&svm->vcpu);
+		hsave->save.cr3    = kvm_read_cr3(vcpu);
 
 	copy_vmcb_control_area(&hsave->control, &vmcb->control);
 
@@ -602,7 +603,7 @@ int nested_svm_vmrun(struct vcpu_svm *svm)
 	nested_svm_vmexit(svm);
 
 out:
-	kvm_vcpu_unmap(&svm->vcpu, &map, true);
+	kvm_vcpu_unmap(vcpu, &map, true);
 
 	return ret;
 }

From patchwork Fri Apr  2 00:43:30 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Krish Sadhukhan <krish.sadhukhan@oracle.com>
X-Patchwork-Id: 12180179
Return-Path: <kvm-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-18.8 required=3.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,HEADER_FROM_DIFFERENT_DOMAINS,INCLUDES_CR_TRAILER,
	INCLUDES_PATCH,MAILING_LIST_MULTI,MSGID_FROM_MTA_HEADER,SPF_HELO_NONE,
	SPF_PASS,USER_AGENT_GIT autolearn=ham autolearn_force=no version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id 4EFC6C43460
	for <kvm@archiver.kernel.org>; Fri,  2 Apr 2021 01:33:06 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id 0D13760FF3
	for <kvm@archiver.kernel.org>; Fri,  2 Apr 2021 01:33:06 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S234184AbhDBBdE (ORCPT <rfc822;kvm@archiver.kernel.org>);
        Thu, 1 Apr 2021 21:33:04 -0400
Received: from userp2130.oracle.com ([156.151.31.86]:47436 "EHLO
        userp2130.oracle.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S233665AbhDBBdC (ORCPT <rfc822;kvm@vger.kernel.org>);
        Thu, 1 Apr 2021 21:33:02 -0400
Received: from pps.filterd (userp2130.oracle.com [127.0.0.1])
        by userp2130.oracle.com (8.16.0.42/8.16.0.42) with SMTP id
 1321ULi0055454;
        Fri, 2 Apr 2021 01:32:58 GMT
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=oracle.com;
 h=from : to : cc :
 subject : date : message-id : in-reply-to : references :
 content-transfer-encoding : content-type : mime-version;
 s=corp-2020-01-29; bh=sJqjwwn/SYPzwAFhYkRmS6LIMupB5Bf29Dd7pLmjO3Q=;
 b=bo5AKKKYNgNYurXtcIu3rU04yRa5WorHKEgu20TyKlpHtKi1trAvkVXecXtofaOLfCDU
 L0I1KI5vIfJ7289Ulw91lan0lru3AU+b8d0NfSqnBs1ytq2Jxfxy3TdKmY3mhBAQS7Bo
 QEAPKkPtMwII+NpzAR52M7j8Pkus4J/+2QXaBoMnjwmZA/UvpzH3lNznPb/F2uzLoQEm
 Q8RC7htmAArG45RLw7rDWr0rbI1NOhCqRVADVMJouhz28hhhKr3gX3qB0+Am6wOluFob
 USLkIu0QSJBhGoZ04YY+VZXhCtGDv7GWoRvbMpR9Sdzj7FCf5i8C63dCl9TBMvp8ucuj xA==
Received: from aserp3020.oracle.com (aserp3020.oracle.com [141.146.126.70])
        by userp2130.oracle.com with ESMTP id 37n30sbg2x-1
        (version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256
 verify=OK);
        Fri, 02 Apr 2021 01:32:58 +0000
Received: from pps.filterd (aserp3020.oracle.com [127.0.0.1])
        by aserp3020.oracle.com (8.16.0.42/8.16.0.42) with SMTP id
 1321V01v019168;
        Fri, 2 Apr 2021 01:32:58 GMT
Received: from nam10-mw2-obe.outbound.protection.outlook.com
 (mail-mw2nam10lp2105.outbound.protection.outlook.com [104.47.55.105])
        by aserp3020.oracle.com with ESMTP id 37n2abyyvf-1
        (version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256
 verify=OK);
        Fri, 02 Apr 2021 01:32:57 +0000
ARC-Seal: i=1; a=rsa-sha256; s=arcselector9901; d=microsoft.com; cv=none;
 b=HBcv30e8V8LnwX9iFwwAdQ7JkfdgSSGhudwvxsOe0Zj8rAWT5HFQrgH03NcCbI7Nwi5dd3h0SI3rrGyiUweBrg25z6rYghf/61n+EREGVziyRtHaEY/Bjzvkp5ExYG+N8k8DIjh8UaH/uwK8WZFnZVEJsi7BH+d4e3+Wl1HDmBUz5YBJJos2zjKrk8KcdbbBglCfEmrqFV7rHuE7eTzQHOZZo5axyjgUU1P90exa1PG0SgwCAG/Lr/CU6Uj69045YMnUkyqGRAY+46ghgscCEbbN3e2kSR9irLU2eM2G9g7A8RpJha35xJTIQiTgRg0XNc+pO7OKWOmkRgg0ji6TKg==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=microsoft.com;
 s=arcselector9901;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=sJqjwwn/SYPzwAFhYkRmS6LIMupB5Bf29Dd7pLmjO3Q=;
 b=FD43ikTo6Ww3MFtRxFl9uPrSLru/0iifd9DQlH7c+h+fTnt0+rE7W61dCEUOk0hTCa8eGr4Rxw9TXgvRt3oP+oy0xXFuDhm8E9E0mT+axQaEYOS5aU6QM20F6kbzQg2Fgkv6eYQQZETLLYq5Fu/7pZ+UPdzHcPJwL55304ON5gS/9vRQPIg66fRIbvK4xfySdtnqCLGByD2zsGtFSmq1Clt//5/li83HQMQbMavsEFLNYDUVO9Kv8YjywT499v6BH6JaNQAmhNFu5IU4x1x1Fko5NWFXvRliPuGPWv42wTDzdJnyrzwtR8at31pyLgMfowpPpKkvvE6jJnztuZgWnQ==
ARC-Authentication-Results: i=1; mx.microsoft.com 1; spf=pass
 smtp.mailfrom=oracle.com; dmarc=pass action=none header.from=oracle.com;
 dkim=pass header.d=oracle.com; arc=none
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=oracle.onmicrosoft.com; s=selector2-oracle-onmicrosoft-com;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=sJqjwwn/SYPzwAFhYkRmS6LIMupB5Bf29Dd7pLmjO3Q=;
 b=HcKPcuOGASz87SLAQLoU6w8NpXabaQjCDaCHGG9uGKkdnUdc8/aPHkkTXRPawoWhg1rnoaNE/siYephtIg/4GjOvvv9ZdEUjXf0ZXg65+j1PL233EB+sTgveh7gKx0GPclaJES9xLiL0piy+wIxlF92Scumy5cpAC0Ga0bUW4xc=
Authentication-Results: vger.kernel.org; dkim=none (message not signed)
 header.d=none;vger.kernel.org; dmarc=none action=none header.from=oracle.com;
Received: from SN6PR10MB3021.namprd10.prod.outlook.com (2603:10b6:805:cc::19)
 by SA2PR10MB4795.namprd10.prod.outlook.com (2603:10b6:806:11d::14) with
 Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.3999.28; Fri, 2 Apr
 2021 01:32:56 +0000
Received: from SN6PR10MB3021.namprd10.prod.outlook.com
 ([fe80::5911:9489:e05c:2d44]) by SN6PR10MB3021.namprd10.prod.outlook.com
 ([fe80::5911:9489:e05c:2d44%5]) with mapi id 15.20.3999.029; Fri, 2 Apr 2021
 01:32:56 +0000
From: Krish Sadhukhan <krish.sadhukhan@oracle.com>
To: kvm@vger.kernel.org
Cc: pbonzini@redhat.com, jmattson@google.com, seanjc@google.com
Subject: [PATCH 4/5 v6] nSVM: Test addresses of MSR and IO permissions maps
Date: Thu,  1 Apr 2021 20:43:30 -0400
Message-Id: <20210402004331.91658-5-krish.sadhukhan@oracle.com>
X-Mailer: git-send-email 2.25.4
In-Reply-To: <20210402004331.91658-1-krish.sadhukhan@oracle.com>
References: <20210402004331.91658-1-krish.sadhukhan@oracle.com>
X-Originating-IP: [138.3.201.29]
X-ClientProxiedBy: SJ0PR13CA0024.namprd13.prod.outlook.com
 (2603:10b6:a03:2c0::29) To SN6PR10MB3021.namprd10.prod.outlook.com
 (2603:10b6:805:cc::19)
MIME-Version: 1.0
X-MS-Exchange-MessageSentRepresentingType: 1
Received: from ban25x6uut29.us.oracle.com (138.3.201.29) by
 SJ0PR13CA0024.namprd13.prod.outlook.com (2603:10b6:a03:2c0::29) with
 Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.4020.9 via Frontend
 Transport; Fri, 2 Apr 2021 01:32:55 +0000
X-MS-PublicTrafficType: Email
X-MS-Office365-Filtering-Correlation-Id: cf0b5b72-5cba-49a5-318d-08d8f5773db6
X-MS-TrafficTypeDiagnostic: SA2PR10MB4795:
X-Microsoft-Antispam-PRVS: 
 <SA2PR10MB47955EF65F550FE6E7CB361B817A9@SA2PR10MB4795.namprd10.prod.outlook.com>
X-MS-Oob-TLC-OOBClassifiers: OLM:486;
X-MS-Exchange-SenderADCheck: 1
X-Microsoft-Antispam: BCL:0;
X-Microsoft-Antispam-Message-Info: 
 7yC+9xeWT1ZF9JxKjmdX/V0EdCj5m0ken34qOo6TSnROBix/72dDmhogVzdNdfmX4EFbUaQHVHzmrFXkektwk+qVlb/XrMpPiMVnQ3h4C0XwIRWJe0ZnN3rMQaUGfgtVae48Y7ZERQcQF6qWhRMUieIKVAoGXUZjXTGrrF+ZFEDFFto+JHH/ZXzu6fhzjqVkYmlVUrBJcoid+pvnsgvKK6Mqiw/EXyq1CCA9qlToT9q/IoD99Z5F6brcQelg07sJaG2hChaQxMcRac3R3dRX42CP9o7vgcbLkSbU0Oox2p7bpD0Xtje2767hLHbyNF731Yx39DTkB/RnZc36QzMx023W8ksQJfp6ApO7fWzH1tr+QsmPI8VJF1Jn874qO6N0sgywIxbiArKfGtDIpLm0vYuSY/NP9ZOiSbwzJ3GHD9T1o1UEI2UHSLms1bPcQ/kYSh/oyo7EBSxVGl3c7R0ZPY1gtFN14zZqImr9/ctmuji7fg9QgjgGvXc/K9y7gqB/P4owIkSZG1F/St96l6nIaZA2dVb7R0PywKnI2W98ijEGTa2K7OiALVB2vSOiaDHVOxAC+APxLAfv17bjjK25y+NpFDvbVWYEnKE+cpDoZReRugivgZJaBEUqoyg2jBzQm1G1jdyY8izEuPsUgAKdfQ==
X-Forefront-Antispam-Report: 
 CIP:255.255.255.255;CTRY:;LANG:en;SCL:1;SRV:;IPV:NLI;SFV:NSPM;H:SN6PR10MB3021.namprd10.prod.outlook.com;PTR:;CAT:NONE;SFS:(376002)(366004)(346002)(396003)(39860400002)(136003)(6666004)(26005)(36756003)(316002)(83380400001)(1076003)(66556008)(5660300002)(16526019)(66476007)(186003)(7696005)(6486002)(2616005)(44832011)(52116002)(956004)(8936002)(38100700001)(8676002)(6916009)(2906002)(66946007)(86362001)(4326008)(478600001);DIR:OUT;SFP:1101;
X-MS-Exchange-AntiSpam-MessageData: 
 go8flqLgj3BzZrMkXrMHY28AlMQulzU52J+ZL69FVeQl39ya3V/6pTBxd2dhjha9Jlt3Dr2lMCxChoC6Vb6JByOZXR0AdfzdGSTcCFgR3v0uJWwJHiTrsbWo6XISy0CBAS3x1VlbqR7Hk2cKpa43FO22TdzpsNpVZWZfY2fC/veAI0PiLczafu19P5A7c3x0QXHyAQxsU8jwlztb6E2tImcWXVafqjkBULFQ4vwcAMuISqOaSV1euTTozFy4oLduoS0DyhqtmEg76RU8uGk/JFuhOT6AOfzxpiphh1DnmPpGEC6d00gv7iM0sh9ho011IFFYf8AH14y3OgPegR9rBokbvUYpjEs1uhZ5ZktFP2k6y/6ShJdPk+PYutAQIFUtqW24mqee68X23Ep+LsxMPgRQ830UG1L/rkdNtqXVlwJKGzry8md0RlJm4DV3P4CSQ4w5gn8eI9bVj4lPoMhbiD7kwgYLw9rrdBUE+1trK8DTqe6gBv9sr25h/qmQLj+gTFu9V1bwP1MmpXoM7YKUYZcytVy6VQMtQSh29Sgf2SbsVPegqpIm8DRUrK7EL5026N0ghD0ZwB9gQdpocMNVX07DYW1CqzKEmVT5uhaZnQTTKx9Y2fYFwGsfWuECWwifY73ffZacGd1rBAOseBXbMpdGyiQnJu01yZKaIKAEMAFAVtARfO+t4Y66A4V8iF3czKNRNk1IvISedL26zW/jUgMZkwtCJSPY/f8Z+QfFCXg95g4U00H2Vwxkx5k1oy9C3TdtcWOju50z1nbUcL0w+CxuuuxV7qSKRIR9eY84g2+WFuRMKA+fen759WqEs4Iv3idZhFA/OYIok37dg/z002A12GGDqNCD9qyaMWKOb8icnIgkY1RRzy08YAMqm0/5qLQuO8gHZk/jDcNGTRlk5xoTUYuXF6fxFrSh+l0UTrknCPkmeJVldme+Is/NScyPqbhqAv1F7iEQJWb7PDPBLdsV+L+PGbXBQE3Y+WFhEBKg0zrmqM/gm0x/TpyUNc6qcTuZdVuCyaOpTJdRIfXgH9/akSjeXx0RzfLogjI2+zP8k8BIcsoPw7haY3jB+LkrSmN1qoXobTKgLJT23nZ6e62haN9J6aSv7HlL4DOMhtzga02HcrprwVd3NIac9GmBc4/k8yp+0z9zUnGtU4LUBb4WJjaexU2EjojCmpclpRfU2rv6qeC+b/D5RLCapx3vOsZFyo2daiwpr0yQjCNxJvHneKObiqfIIXW1R+8zFUDCuaM/SmWn8xng8G74tSK0+aXKNgIt9HgSfmXguGuOxxE5ZofmAqqpUrGXxEzGxTjw1+MzHHKpmxwQeG3GcRs5
X-OriginatorOrg: oracle.com
X-MS-Exchange-CrossTenant-Network-Message-Id: 
 cf0b5b72-5cba-49a5-318d-08d8f5773db6
X-MS-Exchange-CrossTenant-AuthSource: SN6PR10MB3021.namprd10.prod.outlook.com
X-MS-Exchange-CrossTenant-AuthAs: Internal
X-MS-Exchange-CrossTenant-OriginalArrivalTime: 02 Apr 2021 01:32:56.0057
 (UTC)
X-MS-Exchange-CrossTenant-FromEntityHeader: Hosted
X-MS-Exchange-CrossTenant-Id: 4e2c6054-71cb-48f1-bd6c-3a9705aca71b
X-MS-Exchange-CrossTenant-MailboxType: HOSTED
X-MS-Exchange-CrossTenant-UserPrincipalName: 
 35x4o4mV118Mzjs+0rDDbVZqw/4Whbx8VgCejUUJvTHgjGgQQmx3wBsUlcTLpundmffD2eK93sbHvDBWrf8HRWkfBIjzJgojgI0ql4eKOxY=
X-MS-Exchange-Transport-CrossTenantHeadersStamped: SA2PR10MB4795
X-Proofpoint-Virus-Version: vendor=nai engine=6200 definitions=9941
 signatures=668683
X-Proofpoint-Spam-Details: rule=notspam policy=default score=0 adultscore=0
 mlxscore=0 bulkscore=0
 suspectscore=0 phishscore=0 malwarescore=0 mlxlogscore=999 spamscore=0
 classifier=spam adjust=0 reason=mlx scancount=1 engine=8.12.0-2103310000
 definitions=main-2104020008
X-Proofpoint-GUID: MCZIioGIFpc0LLV5AMWGqEwAmZqFf304
X-Proofpoint-ORIG-GUID: MCZIioGIFpc0LLV5AMWGqEwAmZqFf304
X-Proofpoint-Virus-Version: vendor=nai engine=6200 definitions=9941
 signatures=668683
X-Proofpoint-Spam-Details: rule=notspam policy=default score=0
 lowpriorityscore=0 impostorscore=0
 bulkscore=0 priorityscore=1501 suspectscore=0 mlxscore=0 spamscore=0
 clxscore=1015 mlxlogscore=999 malwarescore=0 adultscore=0 phishscore=0
 classifier=spam adjust=0 reason=mlx scancount=1 engine=8.12.0-2103310000
 definitions=main-2104020008
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

According to section "Canonicalization and Consistency Checks" in APM vol 2,
the following guest state is illegal:

    "The MSR or IOIO intercept tables extend to a physical address that
     is greater than or equal to the maximum supported physical address."

Signed-off-by: Krish Sadhukhan <krish.sadhukhan@oracle.com>
(cherry picked from commit 0513cf071255c7d5a1b7a813d017bbdd2d1da263)
Signed-off-by: Krish Sadhukhan <krish.sadhukhan@oracle.com>
---
 x86/svm_tests.c | 42 +++++++++++++++++++++++++++++++++++++++++-
 1 file changed, 41 insertions(+), 1 deletion(-)

diff --git a/x86/svm_tests.c b/x86/svm_tests.c
index 29a0b59..7014c40 100644
--- a/x86/svm_tests.c
+++ b/x86/svm_tests.c
@@ -2304,15 +2304,55 @@ static void test_dr(void)
 	vmcb->save.dr7 = dr_saved;
 }
 
+/*
+ * If the MSR or IOIO intercept table extends to a physical address that
+ * is greater than or equal to the maximum supported physical address, the
+ * guest state is illegal.
+ *
+ * [APM vol 2]
+ */
+static void test_msrpm_iopm_bitmap_addrs(void)
+{
+	u64 saved_intercepts = vmcb->control.intercept;
+	u64 bitmap_addr_1 =
+	    (u64)(((u64)1 << cpuid_maxphyaddr()) - PAGE_SIZE);
+	u64 bitmap_addr_2 =
+	    (u64)(((u64)1 << cpuid_maxphyaddr()) - PAGE_SIZE * 2);
+
+	/*
+	 * MSR bitmap address
+	 */
+	vmcb->control.intercept = saved_intercepts | 1ULL << INTERCEPT_MSR_PROT;
+	vmcb->control.msrpm_base_pa = bitmap_addr_1;
+	report(svm_vmrun() == SVM_EXIT_ERR, "Test MSRPM address: %lx",
+	    bitmap_addr_1);
+	vmcb->control.msrpm_base_pa = bitmap_addr_2;
+	report(svm_vmrun() == SVM_EXIT_ERR, "Test MSRPM address: %lx",
+	    bitmap_addr_2);
+
+	/*
+	 * IOIO bitmap address
+	 */
+	vmcb->control.intercept = saved_intercepts | 1ULL << INTERCEPT_IOIO_PROT;
+	vmcb->control.iopm_base_pa = bitmap_addr_1;
+	report(svm_vmrun() == SVM_EXIT_ERR, "Test IOPM address: %lx",
+	    bitmap_addr_1);
+	vmcb->control.iopm_base_pa = bitmap_addr_2 += 1;
+	report(svm_vmrun() == SVM_EXIT_ERR, "Test IOPM address: %lx",
+	    bitmap_addr_2);
+
+	vmcb->control.intercept = saved_intercepts;
+}
+
 static void svm_guest_state_test(void)
 {
 	test_set_guest(basic_guest_main);
-
 	test_efer();
 	test_cr0();
 	test_cr3();
 	test_cr4();
 	test_dr();
+	test_msrpm_iopm_bitmap_addrs();
 }
 
 

From patchwork Fri Apr  2 00:43:31 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Krish Sadhukhan <krish.sadhukhan@oracle.com>
X-Patchwork-Id: 12180183
Return-Path: <kvm-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-18.8 required=3.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,HEADER_FROM_DIFFERENT_DOMAINS,INCLUDES_CR_TRAILER,
	INCLUDES_PATCH,MAILING_LIST_MULTI,MSGID_FROM_MTA_HEADER,SPF_HELO_NONE,
	SPF_PASS,USER_AGENT_GIT autolearn=ham autolearn_force=no version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id 5D2BAC43600
	for <kvm@archiver.kernel.org>; Fri,  2 Apr 2021 01:33:06 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id 1F40C6108B
	for <kvm@archiver.kernel.org>; Fri,  2 Apr 2021 01:33:06 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S234264AbhDBBdE (ORCPT <rfc822;kvm@archiver.kernel.org>);
        Thu, 1 Apr 2021 21:33:04 -0400
Received: from aserp2130.oracle.com ([141.146.126.79]:49888 "EHLO
        aserp2130.oracle.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S233786AbhDBBdD (ORCPT <rfc822;kvm@vger.kernel.org>);
        Thu, 1 Apr 2021 21:33:03 -0400
Received: from pps.filterd (aserp2130.oracle.com [127.0.0.1])
        by aserp2130.oracle.com (8.16.0.42/8.16.0.42) with SMTP id
 1321VdPv051973;
        Fri, 2 Apr 2021 01:33:00 GMT
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=oracle.com;
 h=from : to : cc :
 subject : date : message-id : in-reply-to : references :
 content-transfer-encoding : content-type : mime-version;
 s=corp-2020-01-29; bh=vSFnUOcVxQ7ReIJxsjGeQiIWjRDiO+g01+LJBKN13XM=;
 b=yr2fV2LySfei8LxDfnlVc5Rgt4sJeSSrQzF4n1TQmm8K1vtlJa8Y9AUJyFncY20DX20+
 ClsOaFoWDNesAk6SNxt1kwI4b/EkdllvqljudQlNACLk85KOkl0TE1Eal7EuA8gmFOEJ
 RcL8AQEWIdgczGi7VXkc/YJr7LI23aouIsIKGN4PlPupE4PIGu//WqRiAh9uNtlPQzNr
 Swg2zk/ni0mJLxb8qCNMO3JrIJXvwpJL3nsMoPUnnIUYfY7k2Pt4Qvt47zz0ClFuG0Hd
 3lmdHyerLrASRKyhmI498u4iH9LkGNKMnizbfZRAuAtPur+IuWJHspj3iCcITQjq9l/t jQ==
Received: from aserp3020.oracle.com (aserp3020.oracle.com [141.146.126.70])
        by aserp2130.oracle.com with ESMTP id 37n33dugmf-1
        (version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256
 verify=OK);
        Fri, 02 Apr 2021 01:33:00 +0000
Received: from pps.filterd (aserp3020.oracle.com [127.0.0.1])
        by aserp3020.oracle.com (8.16.0.42/8.16.0.42) with SMTP id
 1321UwGO018834;
        Fri, 2 Apr 2021 01:33:00 GMT
Received: from nam10-mw2-obe.outbound.protection.outlook.com
 (mail-mw2nam10lp2106.outbound.protection.outlook.com [104.47.55.106])
        by aserp3020.oracle.com with ESMTP id 37n2abyywc-1
        (version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256
 verify=OK);
        Fri, 02 Apr 2021 01:33:00 +0000
ARC-Seal: i=1; a=rsa-sha256; s=arcselector9901; d=microsoft.com; cv=none;
 b=dVyH6iqLlE0oW/B3CrsmeQp7X0Y7j28yxuG1YhVz2RFCSyZePfW0bNeBd6clf2416AhzP1sovrbx9JEwsr4iNihekMl6cCTIUhgugTPpQXnuDQqUXZNqCpW/CZgh6B+3oxDcMNrM21/dBhWZVTCGtAVwxjePWhboyos5A+Pn56Gp3JPxDS9ywlIg0t1d5B67TDkEIY8kcStjBnU1Y+yFm7oLYWdXeQId8cYCAm/ymMCFzr5oNy+NexKRY8Q29aUGcYKNc/42cnW8LqFWz7/04YvxDeTEqv8DKWqe49Lt+DVmUf2ljJGz2SoXoMB5ip3PXQwvqZae6CGlE66EYvsl4w==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=microsoft.com;
 s=arcselector9901;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=vSFnUOcVxQ7ReIJxsjGeQiIWjRDiO+g01+LJBKN13XM=;
 b=dgnXWgupSbzFRLQuVpY5fR8GcL4EDS7ZHhGUWo65ebqo811XnQQMqbFZw/xgs96VMsvfO+nPfUn9D1Hq3zYE+SUoKaBhFhJyOcijmHz/ikcs6bW/iefraiAGgOvMoW7LUYryS1KPggcEIizMDCubrbk/2dSd2PGjukdSRdtFvxbaEO0mnr4WIsyiJ+kWxQBKeYSez0FJacJ9K1wf3BU2cGCE0mCYqlH6TG+fplptgdeee8sU1UYIhK7rcrxyQTuec1j7/WADjImsPuM89W84/0NPw6KJ02gzKa2gTUYm935MT98/B3bYkefIZixJlHJ9g4RZ7u5sP7kxhOXblLawfg==
ARC-Authentication-Results: i=1; mx.microsoft.com 1; spf=pass
 smtp.mailfrom=oracle.com; dmarc=pass action=none header.from=oracle.com;
 dkim=pass header.d=oracle.com; arc=none
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=oracle.onmicrosoft.com; s=selector2-oracle-onmicrosoft-com;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=vSFnUOcVxQ7ReIJxsjGeQiIWjRDiO+g01+LJBKN13XM=;
 b=tNCwTHZA7IYjtf2olDRTyLyLfCl4Y+w8r9Tp2pkfuwDAWaV/AopfSJD52KsTHFIiTWNQLHQd0XrGmj0r7Mu3HjOzyingNLysF5VsEv28iEqG1VrZVSlhVh2nkiTBtVP6MarHVSg5nJ6CN045lMmbBe5BkC8WUd5E5VLc4SQHFL8=
Authentication-Results: vger.kernel.org; dkim=none (message not signed)
 header.d=none;vger.kernel.org; dmarc=none action=none header.from=oracle.com;
Received: from SN6PR10MB3021.namprd10.prod.outlook.com (2603:10b6:805:cc::19)
 by SA2PR10MB4795.namprd10.prod.outlook.com (2603:10b6:806:11d::14) with
 Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.3999.28; Fri, 2 Apr
 2021 01:32:58 +0000
Received: from SN6PR10MB3021.namprd10.prod.outlook.com
 ([fe80::5911:9489:e05c:2d44]) by SN6PR10MB3021.namprd10.prod.outlook.com
 ([fe80::5911:9489:e05c:2d44%5]) with mapi id 15.20.3999.029; Fri, 2 Apr 2021
 01:32:58 +0000
From: Krish Sadhukhan <krish.sadhukhan@oracle.com>
To: kvm@vger.kernel.org
Cc: pbonzini@redhat.com, jmattson@google.com, seanjc@google.com
Subject: [PATCH 5/5 v6] SVM: Use ALIGN macro when aligning 'io_bitmap_area'
Date: Thu,  1 Apr 2021 20:43:31 -0400
Message-Id: <20210402004331.91658-6-krish.sadhukhan@oracle.com>
X-Mailer: git-send-email 2.25.4
In-Reply-To: <20210402004331.91658-1-krish.sadhukhan@oracle.com>
References: <20210402004331.91658-1-krish.sadhukhan@oracle.com>
X-Originating-IP: [138.3.201.29]
X-ClientProxiedBy: SJ0PR13CA0024.namprd13.prod.outlook.com
 (2603:10b6:a03:2c0::29) To SN6PR10MB3021.namprd10.prod.outlook.com
 (2603:10b6:805:cc::19)
MIME-Version: 1.0
X-MS-Exchange-MessageSentRepresentingType: 1
Received: from ban25x6uut29.us.oracle.com (138.3.201.29) by
 SJ0PR13CA0024.namprd13.prod.outlook.com (2603:10b6:a03:2c0::29) with
 Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.4020.9 via Frontend
 Transport; Fri, 2 Apr 2021 01:32:57 +0000
X-MS-PublicTrafficType: Email
X-MS-Office365-Filtering-Correlation-Id: c8d8341d-c877-4da0-f242-08d8f5773f0b
X-MS-TrafficTypeDiagnostic: SA2PR10MB4795:
X-Microsoft-Antispam-PRVS: 
 <SA2PR10MB47958B39C287341DE4BAC193817A9@SA2PR10MB4795.namprd10.prod.outlook.com>
X-MS-Oob-TLC-OOBClassifiers: OLM:305;
X-MS-Exchange-SenderADCheck: 1
X-Microsoft-Antispam: BCL:0;
X-Microsoft-Antispam-Message-Info: 
 2aR9sJw5OxmK0B/yupHGgtyvmzNdNb26TMQwbIz6c/J0bka0H5BxE4jXJ2NWh4gbu2cPGzxJKJMOwcEJLUvK5nRziaLtVd5mlYtQBMDr1JieP2kwDUKlnl+mYLZ5Cqlz8c6mabKZpMB5vevCAATY+RQxU2TDV72GKoV4PvSTjjNIFPf9J+dEir6Z7dlfmNwxebE+2DOolm2Hweo8Bw6KgoDHmzGfS2238x6qaQi55RhafyNZdDkUBARJtwmKqRbroA4Zdwh1/4pubpV5TJFKz6Wqa1lOj+5QCljWERkAMIzwZ69EVJa2/5HgiUAqgHsMn9TqxiG6D4Z6uNg+mTZSuW9V69wPvRRNQv1FXANctmGrmzV0ACmAd9PQWRcon0z6kBo48sWJ5amU6rRsO5YkI2ECWmfE/M1Rij6TJV4nxDbe3bQkAx58aRMmPOk5j9SzYGOVaOwaq8UPR/9298pwJtuVPJrX3MjGkmOd1q/PQsQisOc2xSjgnKhqS1vaOfaJ7E2t1LZidtAX1z9epycOs4N/abjAcsFkkJ3fUTyiJxE9f2c5spyMSHEZc5oN+NTjZpoErVAajUaheq2Vuojz8Y3lPDZBrsLWstexMlQ+KEiiAdASsJUhtcU1SZJPn8rPx6sKbBhwZfeMS8ItD1P5iw==
X-Forefront-Antispam-Report: 
 CIP:255.255.255.255;CTRY:;LANG:en;SCL:1;SRV:;IPV:NLI;SFV:NSPM;H:SN6PR10MB3021.namprd10.prod.outlook.com;PTR:;CAT:NONE;SFS:(376002)(366004)(346002)(396003)(39860400002)(136003)(6666004)(26005)(36756003)(316002)(83380400001)(1076003)(66556008)(5660300002)(16526019)(66476007)(186003)(7696005)(6486002)(2616005)(44832011)(52116002)(956004)(8936002)(38100700001)(4744005)(8676002)(6916009)(2906002)(66946007)(86362001)(4326008)(478600001);DIR:OUT;SFP:1101;
X-MS-Exchange-AntiSpam-MessageData: 
 EY5js1MFva3vyzBj2kvC/E0YPavm83URm2iHdUKqoI8SpctRyq7Hnmwenr7V7MAbA4714Ni1YAp7UsxtP7q7l0aYB81qnjMSp1/PkwQw47b9b6ClALrnqim6xcUqZt00lR1awHRRcJo9ZHo4ix07sBkpfnq3cseCP/NBXqFok4pT6fpwQmeF5huh50ZM1L3Hlk9md493THarUPvh6NiHaVQl1OutsVZYXsICRYlar6ITuAUG5xQIpXDRit8yLFK3n9GypE192+oQjAlmHzW4BGffmpj5HT9SwzY5OtF5+0B4IL6Qq2EOEzym9+m2FFbn7Q4ToRuZgt/eTEbgveLL9M78A9qcCnhRgmFmVBKIslxe4u2LkessG5M7/BQFeFFnunIxFa73Kthz7ZDVkgzHYwq1sC9XKiFb9vMjeUaW3SKFS91t7N2wK/Ut9HH1id3Ad5iq+GVR1CindVxBagTujY2LdcYNXCmsYCr83F3EmAsbkzMai0PbbY05pbMUtZNQbvYD0tf02YAVlBT3A+93CNV25jAnjYPkcu/YRAdRO4T+cui8mPFjL63Ym9vKP2QYeKxb9OE2wvcEj6uDiok+1W1oZeD5jlzMMy84WkSmpVXjtjiHRWJ4scctpK2b8E9HQEb1q6gMpH2RX2BUuiV08yk7WT8khSeVYnt2XLQf30eoLkpWpr5Hy1X3tPpAuNHLvF4K6k82mpw1d/+NS3EIcQVzdOF9vYpoQcl+rUnYJFSdc278DW9XUsXLcMaASEDV7J2QgiCylmu8rDd8qabI5mrrc/9fjZDgAVl2Z1klldn+9bcqxbP8HFOk+f/5dX2CUvff/2cCL9mkNUWj8wkQ1hGNRJhGfsQ6H8XnAxW5eBbOXwAPDm5FCiC6F/j1ws4lOwlo/Co/KfW+9TOlK/gEvk+pcBBS56ZPjtfits3MrBw8PvAgRRyb28q11vbtRVRVc7JIGAqKDONXaa7XNCO8DsS/yqplnkBTGvMo6Ip9Udd+3/0vWq3HKgkC/tdDnm6EGgAGFujQZeqYBhOzyEjIws7vN889lT89Ohvhgy/tzmzpCF/NhCc4OASt7mOviDbmND+eOQaIHEps0gBugXPtmW5Du8ANSKEvOOtrLqBG3C2xzU2IkfGY2uwanmxjNGUxOe0Ah9ISTbXsM9ZzrFE8LCIDP42sZm/vCiQqBt9+yidXtV9xGKiN97NcLguZP0HY/VkHpTYsWcJZyAm+lLBi84pWdExMEZwGtL87ZSDhW5QbCpXmNsrPL0OEONI+jzB4p+fUQGanngW23lDCu0Az5lWXu8xhJJDr8U+z2Xxl9JgY77/5pBzEuBk3KnJXB9Iz
X-OriginatorOrg: oracle.com
X-MS-Exchange-CrossTenant-Network-Message-Id: 
 c8d8341d-c877-4da0-f242-08d8f5773f0b
X-MS-Exchange-CrossTenant-AuthSource: SN6PR10MB3021.namprd10.prod.outlook.com
X-MS-Exchange-CrossTenant-AuthAs: Internal
X-MS-Exchange-CrossTenant-OriginalArrivalTime: 02 Apr 2021 01:32:58.5213
 (UTC)
X-MS-Exchange-CrossTenant-FromEntityHeader: Hosted
X-MS-Exchange-CrossTenant-Id: 4e2c6054-71cb-48f1-bd6c-3a9705aca71b
X-MS-Exchange-CrossTenant-MailboxType: HOSTED
X-MS-Exchange-CrossTenant-UserPrincipalName: 
 pKQ2gfm5tKOhshAUR8lWk4u7OFtKwYmTvOuJNr7FiMzSuSVUpD83BsojLiLFMYABw5VxycoKr8OkYMTWUHtUKCZGwQaqfw24Y15dAg4D14o=
X-MS-Exchange-Transport-CrossTenantHeadersStamped: SA2PR10MB4795
X-Proofpoint-Virus-Version: vendor=nai engine=6200 definitions=9941
 signatures=668683
X-Proofpoint-Spam-Details: rule=notspam policy=default score=0 adultscore=0
 mlxscore=0 bulkscore=0
 suspectscore=0 phishscore=0 malwarescore=0 mlxlogscore=999 spamscore=0
 classifier=spam adjust=0 reason=mlx scancount=1 engine=8.12.0-2103310000
 definitions=main-2104020008
X-Proofpoint-GUID: yELN4fiCMznSoy3RQKVyybAxwmXbbtue
X-Proofpoint-ORIG-GUID: yELN4fiCMznSoy3RQKVyybAxwmXbbtue
X-Proofpoint-Virus-Version: vendor=nai engine=6200 definitions=9941
 signatures=668683
X-Proofpoint-Spam-Details: rule=notspam policy=default score=0 spamscore=0
 mlxlogscore=999 mlxscore=0
 lowpriorityscore=0 suspectscore=0 priorityscore=1501 phishscore=0
 clxscore=1015 impostorscore=0 malwarescore=0 bulkscore=0 adultscore=0
 classifier=spam adjust=0 reason=mlx scancount=1 engine=8.12.0-2103310000
 definitions=main-2104020008
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

Since the macro is available and we already use it for MSR bitmap table, use
it for aligning IO bitmap table also.

Signed-off-by: Krish Sadhukhan <krish.sadhukhan@oracle.com>
(cherry picked from commit bddbbf66215b4fc0076d1391296f11ce3bee7c63)
Signed-off-by: Krish Sadhukhan <krish.sadhukhan@oracle.com>
---
 x86/svm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/x86/svm.c b/x86/svm.c
index a1808c7..846cf2a 100644
--- a/x86/svm.c
+++ b/x86/svm.c
@@ -298,7 +298,7 @@ static void setup_svm(void)
 	wrmsr(MSR_EFER, rdmsr(MSR_EFER) | EFER_SVME);
 	wrmsr(MSR_EFER, rdmsr(MSR_EFER) | EFER_NX);
 
-	io_bitmap = (void *) (((ulong)io_bitmap_area + 4095) & ~4095);
+	io_bitmap = (void *) ALIGN((ulong)io_bitmap_area, PAGE_SIZE);
 
 	msr_bitmap = (void *) ALIGN((ulong)msr_bitmap_area, PAGE_SIZE);
 
